FEATURE: media content modules

coderimus opened the pull request:
### Description (*)
This pull request provides `MediaContentApi` and `MediaContent` modules described at #964 

### Fixed Issues (if relevant)
#964 

### Manual testing scenarios (*)
N/A

coderimus committed:
FEATURE: media content modules

sivaschenko commented on MediaContent/README.md:
```diff
@@ -0,0 +1,13 @@
+# Magento_MediaContent module
+
+The Magento_MediaContent module module provides implementations for managing relations between content and media files used in that content.
```
```suggestion
The Magento_MediaContent module provides implementations for managing relations between content and media files used in that content.
```

sivaschenko commented on MediaContent/README.md:
```diff
@@ -0,0 +1,13 @@
+
+## Extensibility
+
+Extension developers can interact with the Magento_Magento_MediaContent module. For more information about the Magento extension mechanism, see [Magento plug-ins](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/plugins.html).
```
```suggestion
Extension developers can interact with the Magento_MediaContent module. For more information about the Magento extension mechanism, see [Magento plug-ins](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/plugins.html).
```

sivaschenko commented on MediaContentApi/Api/AssignAssetInterface.php:
```diff
@@ -0,0 +1,24 @@
+namespace Magento\MediaContentApi\Api;
+
+/**
+ *
```
Please add a description to the interfaces and interface methods

sivaschenko reviewed (changes requested):
Hi @coderimus , thanks for the PR, please see my review comments

coderimus committed:
Provide method logic and chagnes after the PR review

coderimus committed:
Merge remote-tracking branch 'upstream/1.1-develop' into 964-media-content-api

coderimus commented:
@sivaschenko thank you for your review. I implemented requested changes and also added logic to the newly provided methods.

I have a question: should we return the list of media asset objects from the `\Magento\MediaContent\Model\GetAssetsUsedInContent` or just return the list of raw data from the data storage is applicable for now?

sivaschenko commented:
@coderimus thanks for updates! I think it's better to return the array of media asset ids from GetAssetsUsedInContent

coderimus committed:
Changes according review

sivaschenko committed:
Fixed typos in the readme file

ghost commented:
Hi @coderimus, thank you for your contribution! </br>Please, complete [Contribution Survey](https://opensource.magento.com/survey/contribution/magento+adobe-stock-integration+997), it will take less than a minute. </br>Your feedback will help us to improve contribution process.
